pkg/blobstore/configuration/create_blob_access.go

```diff
@@ -56,6 +59,36 @@ func CreateBlobAccessObjectsFromConfig(configurationFile string) (blobstore.Blob
 	return contentAddressableStorage, actionCache, nil
 }
 
+func newClientTLSFromFiles(config *pb.ClientTLSConfiguration) (grpc_credentials.TransportCredentials, error) {
+	var certificate tls.Certificate
```

Comment:
I think this looks a bit suspicious. In case no client certificate/key is provided, we end up creating a slice with a default-initialized `tls.Certificate`. What if we were to restructure this to:

```go
var certificates []tls.Certificates
if config.ClientCertFile != "" || config.ClientKeyFile != "" {
    certificate, err := tls.LoadX509KeyPair(....)
    if err != nil {
        return nil, err
    }
    certificates = append(certificates, certificate)
}
```

Added advantage of that approach is that you can then remove the `var err error` and `var pemCerts []byte` later on.